lib/3scale/backend/stats/partition_eraser_job.rb

```diff
@@ -0,0 +1,59 @@
+module ThreeScale
+  module Backend
+    module Stats
+      # Job for deleting service stats
+      # Perform actual key deletion from a key partition definition
+      class PartitionEraserJob < BackgroundJob
+        # low priority queue
+        @queue = :stats
+
+        class << self
+          include StorageHelpers
+          include Configurable
+
+          def perform_logged(_enqueue_time, service_id, applications, metrics, users,
+                             from, to, offset, length, context_info = {})
+            job = DeleteJobDef.new(
+              service_id: service_id,
+              applications: applications,
+              metrics: metrics,
+              users: users,
+              from: from,
+              to: to
+            )
+
+            validate_job(job, offset, length)
+
+            stats_key_gen = KeyGenerator.new(job.to_hash)
+
+            stats_key_gen.keys.drop(offset).take(length).each_slice(configuration.stats.delete_batch_size) do |slice|
+              storage.del(slice)
+            end
+
+            [true, { job: job.to_hash, offset: offset, lenght: length }.to_json]
+          rescue Backend::Error => error
+            [false, "#{service_id} #{error}"]
+          end
+
+          private
+
+          def validate_job(job, offset, length)
```

Comment:
Can this raise here?

I think the code is already performing these validations when the request comes through the internal API right? If it finds a problem there it returns an error to the caller so that they can fix the issue.

If we reach this point without returning any error before, the caller will not have a chance to fix the issue. This will just log an error in backend.

Reply:
I think that as an extra protection this might be fine, but if this ever raises, we will need to fix some code in the above layers.

Reply:
this method validates extra params new and specific for this worker job:

`offset` and `length`

They are created by `PartitionGeneratorJob`. Internal API does not receive then

Reply:
Right, I was confused.